### ISSUE TITLE
work with expo

### ISSUE THREAD

**Issue by MauMaxxa** [viromedia/viro#63]
Hello!! wonderful job!

is it possibile to make it work with expo.io?!

Thank you!

**Comment by dam00n** [comment]
Hi MauMaxxa, thank you for the kind words, we hope you are enjoying developing with Viro. Viro has not been integrated with expo.io. Are there any specific features from expo.io that you wish Viro had? Let us know a little more about your needs and we can prioritize accordingly.

**Comment by MauMaxxa** [comment]
Hi dam00n, ant thank for your reply!
Mostly, for its easy set-up of local server (ngrok is very very slow idk why)
By the way, I just tried your example app, and i dont know what am i doing wrong, it always fails to load every OBJ or VRX (which i converted as was written with ViroFBX) i use, except for the ones in the demo.
When something loads, it's always blank without textures, even though i correctly list them as resources, as in your tutorial.

**Comment by dam00n** [comment]
Yeah, ngrok can be slow. You can also try using IP address rather than ngrok from your local server. That works better for me generally. Just type you IP address in the same place you are inputting the ngrok url in the testbed app.

Not sure what the issue is with your FBX/OBJ. Just making sure, you are adding lighting to your scene? If so, do you mind sending over 1 or 2 examples of FBX/OBJ files that are not working and we can test them out. Thanks

**Comment by MauMaxxa** [comment]
Okay i'll try to explain it to you:

- For example, i tried this: https://free3d.com/3d-model/coffee-table-54484.html , and put that into ./js/ARSample/res/test of 
- I downloaded your example here: https://github.com/viromedia/viro
- I change, in HelloWorldSceneAR.js, where you load emoji_smile.vrx with this: 
`<Viro3DObject
              source={require('./res/test/Center Table.obj')}
              position={[0, .2, 0]}
              scale={[.2, .2, .2]}
              type="OBJ"
              lightReceivingBitMask={3}
              shadowCastingBitMask={2}
              highAccuracyGaze={true}
              transformBehaviors={['billboard']}
              resources={[
                         require('./res/test/Center Table.mtl'),
                         require('./res/test/005_wood_001_bump.jpg'),
                         require('./res/test/005_wood_001_color.jpg'),
                         require('./res/test/afrormosia-1_d.jpg'),
                         require('./res/test/book_pages.png'),
                         require('./res/test/mi_cov_ph.jpg'),
                       ]}/>` 

But the object disappear, and i dont get any error..!

Thank you!
Mauro

**Comment by dam00n** [comment]
Hi @MauMaxxa, I was able to replicate the issue on my end. We took a look at the specific OBJ file and it seems like there are some issues with it. We were unable to open it in Maya and other third party 3D modeling programs either had errors or the textures were not appearing. We will continue to look into it to see if there is anything we else can do but for now I would suggest finding another file if possible. 

We have found that 3D models have a wide variance in how they are constructed and not all are compatible with rendering engines. Also, if you are able to find FBX files, those seem to have more consistency in their structure and conversion.

**Comment by MauMaxxa** [comment]
Hello my friend!

I tried a lot of .fbx files, but no-one gaves me textures when i open it in fbx review.
I dont know why.. is there a specific way to export it from maya/other 3d modeling program, so i'll be sure they will work?

Thank you
Mauro

**Comment by dam00n** [comment]
One way we have had some success with is by exporting OBJ/FBX files with Maya using the game export method -> https://knowledge.autodesk.com/support/maya/learn-explore/caas/CloudHelp/cloudhelp/2016/ENU/Maya/files/GUID-2DB6E7B0-04B8-4585-91E9-7D64B02D0338-htm.html

There are some workflows on Unity/game forums about techniques for how to assign materials in blender for FBX files -> http://answers.unity3d.com/questions/124817/why-is-my-imported-fbx-file-not-showing-textures.html

We are less familiar with Blender -> FBX but there are quite a few answers around the web and some Youtube tutorial videos.

**Comment by dam00n** [comment]
Also, if you can send us a link or attach one of the FBX files you are having issues with, we can take a look and perhaps provide more concrete steps for making them work.

**Comment by achuvm** [comment]
Hi @MauMaxxa,

Are you still having issues with your FBX files? If so, feel free them send them our way so we can take a look.

Thanks

Andy

**Comment by MauMaxxa** [comment]
Hi guys
I was having some issues with fbx and maya, will test next week when i'll come back home, so i can test even with your updates!
Thanks!!

**Comment by dam00n** [comment]
Closing due to inactivity

**Comment by WillGeller** [comment]
I am still having issues with the test bed app regardless of using the local ip address or ngrok url. I would love to see support for expo. I think it would help expand the reach of Viro, and make it much easier to share and test on live devices

**Comment by dam00n** [comment]
@WillGeller we are working on improving our testbed app now. If you can let us know which device and os you are having issues with we can add it to our backlog.

Re: Expo - we have native code and our understand is that the Expo team would need to add Viro support into their framework. You can file a feature request with Expo here -> https://expo.canny.io/feature-requests

**Comment by robertjuh** [comment]
Hi,

Sorry for jumping in on this, but i would like to use https://github.com/expo/expo-three inside of the viro-react project. I tried following the instructions but having problems with versions and compatability, is that because Expo has no Viro support OR there's still some version incompatability?

Anyway, would it be possible to use THREEjs inside a Viro project? 
The most known library i could find was: https://github.com/expo/expo-three

**Comment by dam00n** [comment]
@robertjuh Expo does not support Viro. Viro has native code and our understanding is that the Expo team would need to add Viro support into their framework. You can file a feature request with Expo here -> https://expo.canny.io/feature-requests

**Comment by robertjuh** [comment]
Also if i wish to continue using Viro but only use 1 expo 'module'/component ? Or will that never be possible?

**Comment by dam00n** [comment]
Which module/component do you want to use? I am not sure if that is possible but we can look into it.

**Comment by robertjuh** [comment]
We are already talking on the slack and the focus of my problem shifted to using THREEjs in the AR scene and at the time there only seemed to be one plugin/component from expo that could potentially do this, but now that i found 'react-native-gl' and got that succesfully imported in Viro, i'm studying that and hoping to find some possibilities.

**Comment by robertjuh** [comment]
I'll make a new issue on this THREEjs matter, sorry for derailing from the original question

**Comment by WebandMobileMaster** [comment]
Hello, Everyone.
So can not we use the react-viro with the expo?
Should I use it on the react-native-cli?

**Comment by marcofugaro** [comment]
@WebandMobileMaster check out https://github.com/ViroCommunity/starter-kit

**Comment by wodin** [comment]
> Closing this issue. If you would like Expo to support Viro please file a feature request with Expo here -> https://expo.canny.io/feature-requests
> 
> Viro has native code and our understanding is that the Expo team would need to add Viro support into their framework.

They are working on enabling this sort of thing with [Config Plugins](https://docs.expo.io/guides/config-plugins/)

**Comment by markymc** [comment]
@dam00n Just doing some research on AR in React Native and found your repo. Now that https://docs.expo.dev/config-plugins/introduction/ are a standard thing, I wonder if you guys might be willing to create an Expo config plugin?

**Comment by wodin** [comment]
@markymc see this:
https://viro-community.readme.io/docs/integrating-with-expo
